src/components/multi-action-button/multi-action-button.component.tsx

```diff
@@ -198,29 +217,29 @@ export const MultiActionButton = ({
       setTimeout(() => {
         additionalButtons.current[0]?.current?.focus();
       }, 0);
+    } else if (Events.isEscKey(ev)) {
+      ev.preventDefault();
+      setShowAdditionalButtons(false);
     }
   };
 
-  const focusMainButton = () => {
+  const handleFocusMainButton = () => {
```

Comment:
Again not something you've added, and this isn't as clear-cut as the isMainButtonFocused ref (see next comment) - but it makes me uneasy that we are using a ref to tell us if the main button is focused or not. We can tell this whenever we need it by just grabbing document.activeElement and comparing it to the ref we've got for the main button. Might still be better to have the ref if we needed this check in a bunch of places - but it looks to me like it's only once, on line 59 inside hideButtons (and I'm not 100% sure we even need that any more as we no longer want the menu to always open on focus). And it would avoid lots of boilerplate like this handleFocus/handleBlur functions.